### PR TITLE
Add tx-regex, tx-replace, and tx-convert args.

### PR DESCRIPTION
**Pull request by willhite**
Also, make help text more complete.

### CODE REVIEW

**Comment by willhite** [comment]
@mikegray These are the last few changes to nomdex that I'll be making for  a while. The 'tx' args were necessary for the video presentation that I'm doing. They are a bit hacky but get the job done for what I need and may be useful in other cases.

File: samples/go/nomdex/nomdex_find.go [hunk 1]
```diff
@@ -18,42 +18,54 @@ import (
...
+struct Person {
+  name String,
+  geopos struct GeoPos {
+     latitude Number,
+     longitude Number,
+  }
+}
 
+Objects of this type can be indexed on the name, latitude and longitude fields
+with the following commands:
```
Comment (mikegray) [inline samples/go/nomdex/nomdex_find.go]: Does it matter if these are in a List, Set, or Map?  You don't really say what people.value is below - does it need to be a Collection? Might be useful to specify what you are assuming in this example - e.g. "Consider objects of the following type in a Noms list".

File: samples/go/nomdex/nomdex_update.go [hunk 2]
```diff
@@ -23,16 +26,63 @@ import (
...
+   nomdex up --in-path <dsSpec>.value --by .address.city --out-ds personId-index
+
+The previous commands can be understood as follows. The first command updates or
+builds an index by scanning all the objects that are reachable from |in-path| that
+have a string or number value reachable using |by| and stores the root of the
+resulting index in a dataset specified by |out-ds|.
+
+Notice that the --in-path argument has a value of '<dsSpec>.value'. The '.value'
+is not strictly necessary but it's normally useful when indexing. Since datasets
+generally point to Commit objects in NOMS, they usually have parents which are
```
Comment (mikegray) [inline samples/go/nomdex/nomdex_update.go]: Change NOMS to Noms.

File: samples/go/nomdex/nomdex_update.go [hunk 3]
```diff
@@ -126,8 +187,35 @@ func addElementsToGraphBuilder(gb *types.GraphBuilder, db datas.Database, rootOb
...
+			k2 = txRe.ReplaceAllString(string(k1), txReplaceArg)
+		} else {
+			matches := txRe.FindStringSubmatch(string(k1))
+			if len(matches) > 0 {
+				k2 = matches[len(matches)-1]
+			}
+		}
+		if txConvertArg == "number" {
+			if k2 == "" {
+				//fmt.Println("can't convert empty string, returning")
```
Comment (mikegray) [inline samples/go/nomdex/nomdex_update.go]: Remove commented out code.

File: samples/go/nomdex/nomdex_find.go (hunk 1 above)
Comment (mikegray) [inline samples/go/nomdex/nomdex_find.go]: nevermind, i saw the answer below.

**Comment by mikegray** [comment]
lgtm modulo spelling and typo nits

**Pull request merged**